src/Lykke.Job.BlockchainOperationsExecutor.AzureRepositories/TransactionExecutions/CommandHandlerEventRepository.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            var storage = AzureTableStorage<CommandHandlerEventEntity>.Create(
+                connectionString,
+                "CommandHandlerEvents",
+                logFactory);
+
+            var blob = AzureBlobStorage.Create(connectionString);
+
+            return new CommandHandlerEventRepository(typeKeys, storage, blob);
+        }
+
+        public async Task<object> TryGetEventAsync(Guid aggregateId, string commandHandlerId)
+        {
+            var entity = await _storage.GetDataAsync(CommandHandlerEventEntity.GeneratePartitionKey(aggregateId),
+                CommandHandlerEventEntity.GenerateRowKey(commandHandlerId));
+
+            if (entity != null)
+            {
+                if (!_keyTypesDictionary.ContainsKey(entity.EventTypeKey))
+                {
+                    throw new ArgumentException($"Unable to map type with key {entity.EventTypeKey}." +
+                                                "Configured keys: " +
+                                                $"{string.Join(", ", _keyTypesDictionary.Select(p => $"{p.Key}:{p.Value.Name}"))}");
+                }
+
+                var containerName = BuildBlobContainerName(commandHandlerId);
+                var keyName = BuildBlobKeyName(aggregateId, entity.CorrelationId);
+
+                return JsonConvert.DeserializeObject(
+                    await _blob.GetAsTextAsync(containerName, keyName),
+                    _keyTypesDictionary[entity.EventTypeKey]);
+            }
+
+            return null;
+        }
+
+        public async Task<T> InsertEventAsync<T>(Guid aggregateId, string commandHandlerId, T eventData)
+        {
+            if (string.IsNullOrEmpty(commandHandlerId))
+            {
+                throw new ArgumentNullException(nameof(commandHandlerId));
+            }
+
+            if (!_typeKeyDictionary.ContainsKey(typeof(T)))
+            {
+                throw new ArgumentException($"Using {typeof(T).Name} not configured. " +
+                                            "Configured types: " +
+                                            $"{string.Join(", ", _typeKeyDictionary.Select(p => $"{p.Value}:{p.Key.Name}"))}",
+                    nameof(eventData));
+            }
+
+            var correlationId = Guid.NewGuid();
+
+            await SaveBlobEntityAsync(aggregateId, correlationId, commandHandlerId, eventData);
+            
+            await _storage.InsertAsync(new CommandHandlerEventEntity
+            {
+                AggregateId = aggregateId,
+                CommandHandlerId = commandHandlerId,
+                EventTypeKey = _typeKeyDictionary[typeof(T)],
+                PartitionKey = CommandHandlerEventEntity.GeneratePartitionKey(aggregateId),
+                RowKey = CommandHandlerEventEntity.GenerateRowKey(commandHandlerId),
+                CorrelationId = correlationId
+            });
+
+            var result =  await TryGetEventAsync(aggregateId, commandHandlerId);
```

Comment:
Why do we read event object from the storage? We already have the instance here.